src/shared/data-value-mutations/data-value-mutations.js

```diff
@@ -45,19 +54,31 @@ function useSharedDataValueMutation({
 
             // This return value becomes the `context` variable in
             // onError (and onSettled) handlers
-            return { previousQueryData }
+            return { previousQueryData, mutationKey }
         },
 
         // If the mutation fails, use the context returned from onMutate to roll back
         onError: (err, newDataValue, context) => {
+            // this should always be the case, unless a SyntaxError occurs?
+            if (isFetchError(err)) {
+                err = new ApiMutationError(
+                    err,
+                    context.mutationKey,
+                    newDataValue
+                )
+            }
             const { shouldRollback } = handleMutationError(err)
+
             if (shouldRollback) {
```

Comment:
Indeed, I've thought about this as well. After a fair bit of iteration the functionality of this hook has been reduced quite a bit. At first it had logic related to showing alert-bar etc., but we've since decided to not show that. 

Another consideration is that other mutations could use this hook in the future. Since we haven't implemented the error-handling in eg. posting `min-max` values and `comments`. So the idea was that this could be shared, and re-use this hook. In the end, the main reason I kept it is because it was implemented this way, and I built upon it, In the current state you could definitely argue that this would belong in the `onError` in `data-value-mutation`.

Reply:
I actually didn't have a preference between what I said and the opposite as you interpreted it 🙂 (just that I think the side effects of `handleMutationError` feel a bit hidden at the moment). I think maybe just getting `shouldRollback` inside the body of the onError callback and passing it to `handleMutationError` (rather than returning it from there) would make the code a bit clearer, as then the function does as its name suggests (handles a mutation error), whereas right now it's a bit easy to think that there aren't side effects and the function is just figuring out `shouldRollback`. 

If you think that the current structure makes the most sense, that's good by me.